File: petproject/aws/lambda-jvm/src/main/java/com/cramsan/petproject/awslambda/dao/PlantImp.java
package com.cramsan.petproject.awslambda.dao;

import com.cramsan.petproject.appcore.storage.Plant;
import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbAttribute;
import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbBean;
import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbPartitionKey;

@DynamoDbBean
public class PlantImp implements Plant {

    private long id;
    private String scientificName;
    private String imageUrl;

    public PlantImp() { }

    public PlantImp(long id, String scientificName, String imageUrl) {
        this.id = id;
        this.scientificName = scientificName;
        this.imageUrl = imageUrl;
    }

    @DynamoDbPartitionKey
    @Override
    public long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    @Override
    public String getScientificName() {
        return scientificName;
    }

    public void setScientificName(String scientificName) {
        this.scientificName = scientificName;
    }

    @Override
    public String getImageUrl() {
        return imageUrl;
    }

    public void setImageUrl(String imageUrl) {
        this.imageUrl = imageUrl;
    }
}

File: petproject/awslambda-java/src/main/kotlin/com/cramsan/petproject/awslambda/dao/PlantImp.kt
package com.cramsan.petproject.awslambda.dao

import com.cramsan.petproject.appcore.storage.Plant
import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbAttribute
import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbBean
import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbPartitionKey

@DynamoDbBean
class PlantImp() : Plant {

    @get:DynamoDbPartitionKey
    @get:DynamoDbAttribute(value = "id")
    var _id: Long = -1
    @get:DynamoDbAttribute(value = "scientificName")
    lateinit var _scientificName: String
    @get:DynamoDbAttribute(value = "imageUrl")
    lateinit var _imageUrl: String

    constructor(
        id: Long,
        scientificName: String,
        imageUrl: String,
    ) : this() {
        _id = id
        _scientificName = scientificName
        _imageUrl = imageUrl
    }

    override val id: Long = _id
    override val scientificName: String = _scientificName
    override val imageUrl: String = _imageUrl
}
